### PR TITLE
GDPR-78 A few small updates e.g. to display last update details

### PR DESCRIPTION
**Pull request by brightonsbox**

### CODE REVIEW

File: backend/api/dataComplianceApi.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const dataComplianceApiFactory = client => {
 
   const includeETagInBody = response => ({
     ...response.body,
-    etag: response.headers.etag.replace('--gzip', ''),
+    etag: response.headers.etag.replace(/--gzip|W\//gi, ''),
```
Comment (brightonsbox) [inline backend/api/dataComplianceApi.js]: To handle weak etags

File: backend/controllers/retentionReasons.js [hunk 2]
```diff
@@ -1,24 +1,27 @@
+/* eslint-disable consistent-return */
```
Comment (brightonsbox) [inline backend/controllers/retentionReasons.js]: A few helper methods here can return null

File: backend/controllers/retentionReasons.js [hunk 3]
```diff
@@ -1,24 +1,27 @@
...
   const sortReasonsByDisplayOrder = reasons => reasons.sort((r1, r2) => (r1.displayOrder > r2.displayOrder ? 1 : -1))
 
-  const getAgencyDescription = (agencies, agencyId) => {
-    const agency = agencies.find(a => a.agencyId === agencyId)
-    return agency ? agency.description : 'Unknown'
-  }
-
   const getOffenderUrl = offenderNo => `${dpsUrl}offenders/${offenderNo}`
 
+  const formatDate = date => date && formatTimestampToDate(date)
```
Comment (andrewrlee) [inline backend/controllers/retentionReasons.js]: these format methods handle null anyway?

File: backend/controllers/retentionReasons.js (hunk 2 above)
Comment (andrewrlee) [reply backend/controllers/retentionReasons.js]: better to not blanket ignore the whole file as the warnings. These are generally easy to fix, which places in particular?

File: backend/controllers/retentionReasons.js (hunk 3 above)
Comment (andrewrlee) [reply backend/controllers/retentionReasons.js]: e.g: https://github.com/ministryofjustice/prisonstaffhub/pull/882/files#diff-d2bd71a8ef7316942b1e5a4166909718L140

Comment (brightonsbox) [reply backend/controllers/retentionReasons.js]: Ah right you are.  For some reason I'd convinced myself that they wouldn't handle it!

File: backend/controllers/retentionReasons.js (hunk 2 above)
Comment (brightonsbox) [reply backend/controllers/retentionReasons.js]: I'll have another look, the linter wasn't having various different arrangements of one function.

**Pull request merged**